### PR TITLE
Fix multiple credentials in route53

### PR DESCRIPTION
**Pull request by yongtang**
### 1. Why is this pull request needed and what does it do?

This fix tries to to address the issue in #2205 while multiple credentials in route53 plugin will be overwritten by previous ones.

### 2. Which issues (if any) are related?

#2205 

### 3. Which documentation changes (if any) need to be made?

n/a

### CODE REVIEW

**Comment by dilyevsky** [comment]
Hm does this actually fix this? Not sure if chained provider is intelligent enough to pick an appropriate credentials for the zone. I think it might just try the first one set and fail. aws docs aren't super clear on that...

**Comment by yongtang** [comment]
I tried with `non-existing provider + valid provider` and it looks to be working. Though whether or not it goes to zone level is unknown. I will try to see if I could setup different credentials with different account to verify.

File: plugin/route53/setup.go [hunk 1]
```diff
@@ -44,8 +44,7 @@ func setup(c *caddy.Controller, f func(*credentials.Credentials) route53iface.Ro
 	// With that said, even though a user doesn't define any credentials in
 	// Corefile, we should still attempt to read the default credentials file,
 	// ~/.aws/credentials with the default profile.
-	sharedProvider := &credentials.SharedCredentialsProvider{}
-	var providers []credentials.Provider
+	providers := []credentials.Provider{&credentials.EnvProvider{}, &credentials.SharedCredentialsProvider{}}
```
Comment (canthefason) [inline plugin/route53/setup.go:47]: Passing an empty SharedCredentialsProvider ends up with incorrect credentials. This is the message I get: 
```
Error during parsing: failed to create Route53 plugin: ExpiredToken: The security token included in the request is expired"
```

**Comment by canthefason** [comment]
I was able to test this out and unfortunately, it didn't work. There are two reasons for it. The first one is the empty SharedCredentialsProvider that I already left a comment ends up with incorrect credentials which is an easy fix. The second reason is, when we add multiple shared credential providers [ListHostedZonesByNameWithContext](https://github.com/coredns/coredns/blob/master/plugin/route53/route53.go#L54) uses the very first shared credentials we pass and as expected, errors out when we query a hosted zone that belongs to the other accounts.

**Comment by canthefason** [comment]
Besides the credentials file, this should also be an issue for static credentials, as well. IMHO, unless AWS clients allow switching between credentials, we need to initialize multiple clients with different sessions instead of passing one for each top-level stanza.

**Comment by yongtang** [comment]
@canthefason Yes I think the per host/session might make sense. Will take a look and see if I could update the PR>

**Comment by yongtang** [comment]
Has been flooded by other things recently, still working on it. Will update (hopefully) soon.

**Comment by miekg** [comment]
closing this for the time being

**Pull request closed without merging**